pegen/pegen.c

```diff
@@ -799,14 +799,33 @@ Pegen_Compare(Parser *p, expr_ty expr, asdl_seq *pairs)
 
 /* Accepts a load name and creates an identical store name */
 expr_ty
-store_name(Parser *p, expr_ty load_name)
+set_store_context(Parser *p, expr_ty node)
 {
-    if (!load_name) {
+    if (!node) {
         return NULL;
     }
-    return _Py_Name(load_name->v.Name.id,
-                    Store,
-                    EXTRA_EXPR(load_name, load_name));
+
+    switch(node->kind) {
+        case Name_kind:
+            return _Py_Name(node->v.Name.id,
```

Comment:
Do we still need to Py_INCREF here?

Reply:
Unclear, I still need to understand how this "borrow" works but I suspect it *may* be ok because the arena is holding it alive and that lives as long as the ast nodes live.